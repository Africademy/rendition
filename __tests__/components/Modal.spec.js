/* globals expect, test */
import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import {
  Modal,
  Provider
} from '../../src'

// react-test-renderer does not play nicely with ReactDOM.createPortal(), which
// is used by Grommet's Layer component in the Modal component. To get it to
// work without throwing an error, the ReactDOM.createPortal() method is stubbed
// here.
// see: https://github.com/facebook/react/issues/11565
ReactDOM.createPortal = node => node

test('Modal renders correctly', () => {
  const component = renderer.create(
    <Provider>
      <Modal>Modal</Modal>
    </Provider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
