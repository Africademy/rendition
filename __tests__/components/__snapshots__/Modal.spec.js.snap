// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  className="sc-bdVaJa iLLxmY"
>
  <div
    className="sc-chPdSV sc-ckVGcZ cLHgGt sc-kAzzGY ikexOF sc-cSHVUG gbsWRV"
    onClick={[Function]}
  >
    <div
      className="sc-jKJlTe cXwbpL sc-jzJRlG cvFLEa sc-fjdhpX fNNSIL"
      z={8888}
    />
    <div
      className="sc-kEYyzF iSkjgM sc-kAzzGY ikexOF sc-cSHVUG gbsWRV"
    >
      <div
        className="sc-kkGfuU bBINRA sc-kAzzGY fLGkVM sc-cSHVUG gbsWRV"
        onClick={[Function]}
      >
        Modal
        <div
          className="sc-chPdSV gWTjoX sc-kAzzGY cEhHZr sc-cSHVUG gbsWRV"
        >
          <button
            className="sc-VigVT dzpHnm sc-htoDjs hQVMTw"
          >
            OK
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
